lib/games_manager.rb

```diff
@@ -92,43 +83,20 @@ def get_season_results(team_id)
     @games.each do |game|
       if team_id == game.away_team_id || team_id == game.home_team_id
         summary[game.season] = [] if summary[game.season].nil?
-        if team_id == game.home_team_id && game.home_goals > game.away_goals
-          result = "W"
-        elsif team_id == game.away_team_id && game.home_goals < game.away_goals
-          result = "W"
-        elsif game.home_goals == game.away_goals
-          result = "T"
-        else
-          result = "L"
-        end
+        result = "non-win"
+        result = "W" if (team_id == game.home_team_id && game.home_goals > game.away_goals) || (team_id == game.away_team_id && game.home_goals < game.away_goals)
         summary[game.season] << result
       end
     end
     summary
   end
 
   def best_season(team_id)
-    most = 0
-    best = nil
-    get_season_results(team_id).each do |key, value|   #refactor-able? #I think we can use .valuses.max/.min
-      if value.count('W').to_f / value.size > most
-        most = value.count('W').to_f / value.size
-        best = key
-      end
-    end
-    best
+    (get_season_results(team_id).max_by {|k, v| v.count('W').to_f / v.size})[0]
```

Comment:
I like that it's one line except I have a really hard time reading it.

Reply:
I share that concern, if it helps we can set it equal to 'best = (get_season_results(team_id).max_by {|k, v| v.count('W').to_f / v.size})[0]'